### ISSUE TITLE
Wrong value from luaH_getn

### ISSUE THREAD

**Issue by unkernet** [fengari-lua/fengari#162]
There is difference between fengari and Lua 5.3:
```lua
local t = {1, nil, 3, 4}
print(#t) -- 1 (4 in Lua 5.3)
```

```lua
local t = {nil, nil, 3, 4, 5, 6, 7, 8}
print(#t) -- 6 Why?
```
According to the Lua 5.3 documentation
> When t is not a _sequence_, #t can return any of its borders.

**Comment by giann** [comment]
Fengari's implementation of the table is different from PUC Rio, but I think both results are valid under the manual's definition.

**Comment by daurnimator** [comment]
> ```lua
> local t = {nil, nil, 3, 4, 5, 6, 7, 8}
> print(#t) -- 6 Why?
> ```

That's a bug. It should be `0` or `8`.

**Comment by daurnimator** [comment]
Please review the fix in #163

**Comment by unkernet** [comment]
Looks good
